src/Microsoft.AspNetCore.Cors/Infrastructure/CorsService.cs

```diff
@@ -78,12 +79,23 @@ public CorsResult EvaluatePolicy(HttpContext context, CorsPolicy policy)
                 throw new ArgumentNullException(nameof(policy));
             }
 
-            var corsResult = new CorsResult();
-            var accessControlRequestMethod = context.Request.Headers[CorsConstants.AccessControlRequestMethod];
-            if (string.Equals(context.Request.Method, CorsConstants.PreflightHttpMethod, StringComparison.OrdinalIgnoreCase) &&
-                !StringValues.IsNullOrEmpty(accessControlRequestMethod))
+            var origin = context.Request.Headers[CorsConstants.Origin];
+            // A CORS-preflight request ... uses `OPTIONS` as method and includes these headers:
+            // Access-Control-Request-Method, Access-Control-Request-Headers
+            // To keep parity with legacy behavior, we'll ignore the presence of Access-Control-Request-Headers header. 
+            var requestHeaders = context.Request.Headers;
+            var isPreflightRequest =
+                string.Equals(context.Request.Method, CorsConstants.PreflightHttpMethod, StringComparison.OrdinalIgnoreCase) &&
+                requestHeaders.ContainsKey(CorsConstants.AccessControlRequestMethod);
+
+            var corsResult = new CorsResult
+            {
+                IsPreflightRequest = isPreflightRequest,
+                IsCorsResponseAllowed = IsOriginAllowed(policy, origin),
+            };
+
+            if (isPreflightRequest)
             {
-                _logger?.IsPreflightRequest();
```

Comment:
Why are we removing logging from here?

Reply:
Moved it to `IsOriginAllowed` which is where we log if the policy is failed